### PR TITLE
statetest: Validate EOF containers in test state

### PR DESCRIPTION
**Pull request by rodiazet**
Part of #585.

### CODE REVIEW

File: test/statetest/statetest_runner.cpp [hunk 1]
```diff
@@ -25,6 +26,14 @@ void run_state_test(const StateTransitionTest& test, evmc::VM& vm)
             const auto tx = test.multi_tx.get(expected.indexes);
             auto state = test.pre_state;
 
+            for (const auto& [add, acc] : state.get_accounts())
```
Comment (chfast) [inline test/statetest/statetest_runner.cpp:29]: Add a comment for these block of code.
Enclose statements that are more than single line with `{}`.

File: test/statetest/statetest_runner.cpp [hunk 2]
```diff
@@ -25,6 +26,14 @@ void run_state_test(const StateTransitionTest& test, evmc::VM& vm)
...
+                if (is_eof_container(acc.code))
+                    if (const auto result = validate_eof(rev, acc.code);
+                        result != EOFValidationError::success)
+                        throw std::invalid_argument(
+                            "deployed contract code validation failure: " + hex0x(add) + ": " +
```
Comment (chfast) [inline test/statetest/statetest_runner.cpp:34]:
```suggestion
                            "EOF container at " + hex0x(add) + " is invalid: " +
```

File: test/statetest/statetest_runner.cpp [hunk 3]
```diff
@@ -25,6 +26,14 @@ void run_state_test(const StateTransitionTest& test, evmc::VM& vm)
...
+                            get_error_message(result).data());
```
Comment (chfast) [inline test/statetest/statetest_runner.cpp:35]:
```suggestion
                            std::string{get_error_message(result)});
```

**Review by chfast — CHANGES_REQUESTED** [review]

File: test/statetest/statetest_runner.cpp (hunk 1 above)
Comment (rodiazet) [reply test/statetest/statetest_runner.cpp:31]: done

File: test/statetest/statetest_runner.cpp (hunk 2 above)
Comment (rodiazet) [reply test/statetest/statetest_runner.cpp]: done

File: test/statetest/statetest_runner.cpp [hunk 4]
```diff
@@ -25,6 +26,22 @@ void run_state_test(const StateTransitionTest& test, evmc::VM& vm)
             const auto tx = test.multi_tx.get(expected.indexes);
             auto state = test.pre_state;
 
+            // Validate deployed EOF container before running state test.
+            // This validation does not pass if state test in invalid only.
```
Comment (chfast) [inline test/statetest/statetest_runner.cpp:30]:
```suggestion
            // Any invalid EOF in state make the test incorrect.
```

File: test/statetest/statetest_runner.cpp [hunk 5]
```diff
@@ -25,6 +26,22 @@ void run_state_test(const StateTransitionTest& test, evmc::VM& vm)
...
+            for (const auto& [add, acc] : state.get_accounts())
```
Comment (chfast) [inline test/statetest/statetest_runner.cpp:31]:
```suggestion
            for (const auto& [addr, acc] : state.get_accounts())
```

**Review by chfast — CHANGES_REQUESTED** [review]
We should also add one test to https://github.com/ethereum/evmone/blob/master/test/integration/statetest/CMakeLists.txt.

File: test/statetest/statetest_runner.cpp [hunk 6]
```diff
@@ -25,6 +26,22 @@ void run_state_test(const StateTransitionTest& test, evmc::VM& vm)
             const auto tx = test.multi_tx.get(expected.indexes);
             auto state = test.pre_state;
 
+            // Validate deployed EOF container before running state test.
```
[the remark this answers is not in the document]
Comment (rodiazet) [reply test/statetest/statetest_runner.cpp:29]: done

File: test/statetest/statetest_runner.cpp (hunk 4 above)
Comment (rodiazet) [reply test/statetest/statetest_runner.cpp:30]: done

File: test/statetest/statetest_runner.cpp (hunk 5 above)
Comment (rodiazet) [reply test/statetest/statetest_runner.cpp:31]: done

**Review by chfast — APPROVED** [review]

**Pull request merged**